packages/amplify_api_rest/android/src/main/kotlin/com/amazonaws/amplify/amplify_api_rest/AmplifyApiRestPlugin.kt

```diff
@@ -0,0 +1,311 @@
+  private var operationsMap : HashMap<String, RestOperation> = HashMap()
+
+  override fun onAttachedToEngine(@NonNull flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
+    channel = MethodChannel(flutterPluginBinding.getFlutterEngine().getDartExecutor(), "com.amazonaws.amplify/amplify_api_rest")
+    channel.setMethodCallHandler(this)
+    Amplify.addPlugin(AWSApiPlugin())
+  }
+
+  companion object {
+    val LOG = Amplify.Logging.forNamespace("amplify:flutter:api_rest")
+
+    @JvmStatic
+    fun registerWith(registrar: Registrar) {
+      val channel = MethodChannel(registrar.messenger(), "com.amazonaws.amplify/amplify_api_rest")
+      channel.setMethodCallHandler(AmplifyApiRestPlugin())
+      Amplify.addPlugin(AWSApiPlugin())
+      LOG.info("Added AWSApiPlugin")
+    }
+
+  }
+
+  override fun onDetachedFromEngine(@NonNull binding: FlutterPlugin.FlutterPluginBinding) {
+    channel.setMethodCallHandler(null)
+  }
+
+  @VisibleForTesting
+  override fun onMethodCall(@NonNull call: MethodCall, @NonNull result: Result) {
+    
+    if(call.method == "cancel"){
+      onCancel(result, (call.arguments as String))
+      return
+    }
+    
+    var arguments : Map<String, Any>
+
+    try {
+      arguments = checkArguments(call.arguments as Map<String, Any>)
+    }
+    catch(e: java.lang.Exception){
+      result.error("WrongMapInput", "The provided map input was not correct", e.message)
+      return
+    }
+    
+    when (call.method) {
+      "get" -> onGet(result, arguments)
+      "post" -> onPost(result, arguments)
+      "put" -> onPut(result, arguments)
+      "delete" -> onDelete(result, arguments)
+      else -> result.notImplemented()
+    }
+  }
+
+  private fun restFunctionHelper(
+          @NonNull flutterResult: Result,
+          @NonNull request: Map<String, *>,
+          @NonNull function3: KFunction3<
+                  @ParameterName(name = "restOptions") RestOptions,
+                  @ParameterName(name = "restConsumer") Consumer<RestResponse>,
+                  @ParameterName(name = "exceptionConsumer") Consumer<ApiException>,
+                  RestOperation?>,
+          @NonNull function4: KFunction4<
+                  @ParameterName(name = "apiName") String,
+                  @ParameterName(name = "restOptions") RestOptions,
+                  @ParameterName(name = "restConsumer") Consumer<RestResponse>,
+                  @ParameterName(name = "exceptionConsumer") Consumer<ApiException>,
+                  RestOperation?>
+  ){
+    if(FlutterRestInputs.validate(request)) {
+      val inputs = FlutterRestInputs(request)
+      val code = inputs.getCode()
+      val apiName: String? = inputs.getApiPath()
+      val options: RestOptions = FlutterRestInputs(request).getRestOptions()
+
+      try {
+
+        if(apiName == null){
+          var operation : RestOperation? = function3(options,
+                  Consumer { result -> prepareRestResponseResult(flutterResult, result, code) },
```

Comment:
I think you ought to be able to omit the `Consumer ` bit here, in all cases, no? Generally you can use just `{ insideAClosure() }`.

Reply:
hmm how do you recommend doing that @jamesonwilliams ?

![image](https://user-images.githubusercontent.com/11478656/102833800-fc369600-43a6-11eb-9a40-7949bc1bae70.png)

Reply:
What does it say when you click on the red lightbulb? The `function3` thing is weird though. I guess it doesn't have  the `Consumer` type built into its signature, and that's why it can't infer the type. Oh well.